tools/swift-format/Sources/Rules/BlankLineBetweenMembers.swift

```diff
@@ -17,5 +17,148 @@ import SwiftSyntax
+        return true
+      }
+    }
+    return false
+  }
+  
+  /// Returns the given trivia without any set of consecutive blank lines
+  /// that exceeds the maximumBlankLines.
+  func removeExtraBlankLines(_ trivia: Trivia, _ member: MemberDeclListItemSyntax) -> Trivia {
+    let maxBlankLines = context.configuration.maximumBlankLines
+    var pieces = [TriviaPiece]()
+    
+    // Iterates through the trivia, verifying that the number of blank
+    // lines in the file do not exceed the maximumBlankLines. If it does
+    // a lint error is raised.
+    for piece in trivia {
+      if case .newlines(let num) = piece,
+         num - 1 > maxBlankLines {
+        pieces.append(.newlines(maxBlankLines + 1))
+        diagnose(.removeBlankLines(count: num - maxBlankLines)
+                 ,on: member)
+      }
+      else {
+        pieces.append(piece)
+      }
+    }
+    return Trivia.init(pieces: pieces)
+  }
+  
+  /// Indicates if a declaration has to be ignored by checking if it's
+  /// a single line and if the format is configured to ignore single lines.
+  func ignoreItem(item: MemberDeclListItemSyntax) -> Bool {
+    guard let firstToken = item.firstToken else { return false }
+    guard let lastToken = item.lastToken else { return false }
+    
+    let isSingleLine = firstToken.positionAfterSkippingLeadingTrivia.line ==
+                       lastToken.positionAfterSkippingLeadingTrivia.line
+    
+    let ignoreLine = context.configuration.blankLineBetweenMembers
+                     .ignoreSingleLineProperties
+    
+    return isSingleLine && ignoreLine
+  }
+  
+  /// Recursively ensures all nested member types follows the BlankLineBetweenMembers rule.
+  func checkForNestedMembers(_ member: MemberDeclListItemSyntax) -> MemberDeclListItemSyntax {
+    if let nestedEnum = member.decl as? EnumDeclSyntax {
+      let nestedMembers = visit(nestedEnum.members)
+      let newDecl = nestedEnum.withMembers(nestedMembers as? MemberDeclBlockSyntax)
+
+      return member.withDecl(newDecl)
+    }
+    else if let nestedStruct = member.decl as? StructDeclSyntax {
+      let nestedMembers = visit(nestedStruct.members)
+      let newDecl = nestedStruct.withMembers(nestedMembers as? MemberDeclBlockSyntax)
+
+      return member.withDecl(newDecl)
+    }
+    else if let nestedClass = member.decl as? ClassDeclSyntax {
+      let nestedMembers = visit(nestedClass.members)
+      let newDecl = nestedClass.withMembers(nestedMembers as? MemberDeclBlockSyntax)
+      
+      return member.withDecl(newDecl)
+    }
+    else if let nestedExtension = member.decl as? ExtensionDeclSyntax {
+      let nestedMembers = visit(nestedExtension.members)
+      let newDecl = nestedExtension.withMembers(nestedMembers as? MemberDeclBlockSyntax)
+      
+      return member.withDecl(newDecl)
+    }
+    
+    return member
+  }
+}
+
+/// Indicates if the given trivia piece is any type of comment.
+func isComment(_ triviaPiece: TriviaPiece) -> Bool {
```

Comment:
This looks like it would make a good extension property on `TriviaPiece`, if it doesn't already exist.